Change_Flashers/Soave_Flasher_OOP.py

```diff
@@ -168,45 +194,50 @@ def func_opt(parametrs):
     P_bar_plot = []
     Fraction_R22_plot = []
     Fraction_R115_plot = []
-    for j in range(len(P_bar[i])):
-        flash_data = calculate_flash(T, P_bar, zs, i, j)
-        # Добавление точек в листы
-        # Реализуемая модель
-        R22_plot.append(flash_data[0][0])
-        R115_plot.append(flash_data[0][1])
-        P_bar_plot.append(P_bar[i][j])
-        # Экспериментальные (исходные) данные
-        Fraction_R22_plot.append(Fraction_R22[i][j])
-        Fraction_R115_plot.append(Fraction_R115[i][j])
-
-        # Абсолютные ошибки
-        ABS_error_R22 = abs(Fraction_R22[i][j] - flash_data[0][0])
-        ABS_error_R115 = abs(Fraction_R115[i][j] - flash_data[0][1])
-
-        # Относительные ошибки
-        Rel_error_R22 = ABS_error_R22 / flash_data[0][0]
-        Rel_error_R115 = ABS_error_R115 / flash_data[0][1]
-
-
-        # Счетчик точек
-        count += 1
-        countT += 1
-        sum_R22T += ABS_error_R22
-        sum_R115T += ABS_error_R115
-
-        sum_R22 += ABS_error_R22
-        sum_R115 += ABS_error_R115
-
-        print('==============================================================================')
-        print('Температура расчета:', T[i], '°C; Давление расчета:', P_bar[i][j], 'Бар')
-        print('Соотношение компонентов смеси R-22, R-115, ISO 32:', flash_data[0])
-        print('Пар ', flash_data[3], '\n')
-        print('****** Абсолютная погрешность ******')
-        print('Погрешность для R-22:', ABS_error_R22)
-        print('Погрешность для R-115:', ABS_error_R115)
-        print('****** Относительная погрешность ******')
-        print('Погрешность для R-22:', Rel_error_R22)
-        print('Погрешность для R-115:', Rel_error_R115)
+    for j in range(len(pressure.values[i])):
+        # Вызов функции для получения мольных долей, температуры и давления
+        flash_data = calculate_flash(T, pressure.values, zs, i, j)
+        if flash_data[3] == 1:

```

Comment:
it is GOOD that you check if there only vapour.
But certainly you need somewhere to increase absolute error